### DISCUSSION TITLE
Suggestion: RON for configuration instead of JSON

### DISCUSSION THREAD

**Discussion by xpe** [zed-industries/zed#6620] [Feature Requests]
Why not [RON] (Rusty Object Notation) for Zed's configuration files? Zed is mostly/all Rust, right?

Apologies if I'm stating the obvious -- or alternately, trying to push a rope uphill -- my suggestion is: pick the better format (for Rust, RON is the winner). Also, there no need to fall into the trap of thinking e.g. "JSON is good enough" or "people are used to JSON; stick to that". RON isn't scary. Plus, it is nice and sane.

[RON]: https://github.com/ron-rs/ron

P.S. Beating VS Code, IMO, can be done. But it won't be done overnight, and it won't be done by any one feature. Instead, it will be by the sustained application of 1% improvements. (idea stolen / paraphrased from the book "Atomic Habits"). The little improvements add up. Relentlessly pick the better way, even if these better things seem small or unconventional.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
